webapp/src/components/AuctionPage/AuctionPage.js

```diff
@@ -168,6 +184,22 @@ export default class AuctionPage extends React.PureComponent {
     return landRegistry.ownerOf(tokenId)
   }
 
+  updateOwner = async parcel => {
+    return this.getParcelOwnerOnChain(parcel).then(ownerOnChain => {
+      if (!Contract.isEmptyAddress(ownerOnChain)) {
+        this.props.onSetParcelOnChainOwner(parcel.id, ownerOnChain)
+      }
+    })
+  }
+
+  updateSelectionOwners = async () => {
+    await Promise.all(this.getSelectedParcels().map(this.updateOwner))
+    setTimeout(() => {
```

Comment:
_Question_: maybe we sould store this timeoutId and clear it on unmount, even if it doesn't do anything with the `if (!this.mounted) return`. I'm thinking it might be cumbersome if the user enters and leaves the /auction page repeatedly

Reply:
✅ done https://github.com/decentraland/marketplace/pull/710/commits/23d917a67410008d925d134506142de74ec61a12